### ISSUE TITLE
Export to s3

### ISSUE THREAD

**Issue by tpharaoh** [pascalbaljetmedia/laravel-ffmpeg#124]
I have a working exportForHLS, but the resulting upload is just the m3u8 file when I use s3. Then it doesn't play. 

When I use local disk, there is the m3u8 and 3 ts files, and it plays.

```
        $file = storage_path() . '/uploads/videos/' . $this->filename;

        $lowBitrate = (new X264)->setKiloBitrate(250);
        $lowBitrate = ( new X264 )->setKiloBitrate( 144 )->setVideoCodec( 'libx264' )->setAudioCodec( "libmp3lame" )->setAdditionalParameters( [
            "-strict",
            "-2",
            ] );
        $midBitrate = (new X264)->setKiloBitrate(500)->setVideoCodec( 'libx264' )->setAudioCodec( "libmp3lame" )->setAdditionalParameters( [
            "-strict",
            "-2",
            ] );;
        $highBitrate = (new X264)->setKiloBitrate(1000)->setVideoCodec( 'libx264' )->setAudioCodec( "libmp3lame" )->setAdditionalParameters( [
            "-strict",
            "-2",
            ] );;
        
        FFMpeg::fromDisk('uploads')
            ->open('videos/'.$this->filename)
            ->exportForHLS()
            ->setSegmentLength(10) // optional
            // ->inFormat(new \FFMpeg\Format\Video\X264('libmp3lame'))
            ->addFormat($lowBitrate)
    
            ->addFormat($midBitrate)
            ->addFormat($highBitrate)
            ->toDisk('s3video')
            ->save('/videos/ffmpeg/'.$this->filename.'.m3u8');

```

**Comment by wikaws** [comment]
i faced the same problem. i've been searching for a day and found no solution. Please let me know if you have any solutions.

**Comment by pascalbaljet** [comment]
As written in the [documentation](https://github.com/pascalbaljetmedia/laravel-ffmpeg#hls), it's currently only possible to export to local disks when doing HLS. A workaround could be exporting the video to an empty (local) folder and then uploading each file in the folder to S3.

**Comment by alexander-rieder** [comment]
are there plans to make the hls export available to s3 storage?

**Comment by pascalbaljet** [comment]
Could you try and test this branch (still WIP, needs some refactoring):
https://github.com/pascalbaljetmedia/laravel-ffmpeg/tree/hls-external

**Comment by alexander-rieder** [comment]
> Could you try and test this branch (still WIP, needs some refactoring):
> https://github.com/pascalbaljetmedia/laravel-ffmpeg/tree/hls-external

how can i test it?

**Comment by alexander-rieder** [comment]
I just tried.... Unfortunately this doesnt seem to work.. only the playlist file is saved to the s3 storage... Do you need further information / debug infos?

**Comment by pascalbaljet** [comment]
I've rewritten the package and support for exporting to S3 (and other external disks) is included. Could you test the `v7` branch? Almost no breaking changes except for some filter stuff in HLS exports.

https://github.com/pascalbaljetmedia/laravel-ffmpeg/tree/v7

`composer require "pbmedia/laravel-ffmpeg:v7.x-dev"`

**Comment by pascalbaljet** [comment]
Fixed in `v7.0`
